src/ray/pubsub/subscriber.cc

```diff
@@ -93,9 +93,10 @@ void SubscriberChannel<KeyIdType>::HandlePublishedMessage(
 
   auto maybe_subscription_callback = GetSubscriptionCallback(publisher_address, key_id);
   if (maybe_subscription_callback.has_value()) {
-    // If the object id is still subscribed, invoke a subscription callback.
-    const auto &subscription_callback = maybe_subscription_callback.value();
-    subscription_callback(pub_message);
+    // If the object id is still subscribed, return a subscribe callback.
+    return maybe_subscription_callback.value();
```

Comment:
if you'd like to avoid capture copying, you'll want to do:

`return std::move(maybe_subscription_callback).value()`